app/controllers/links_controller.rb

```diff
@@ -2,41 +2,48 @@
 
 class LinksController < ApplicationController
   before_action :load_dependencies
+  before_action :set_back_url_before_post_request, only: [:edit, :update, :destroy]
```

Comment:
this might be more succinctly expressed as
`before_action :set_back_url, if: -> { request.post? }`

NBD though, only change it if you want to.